### DISCUSSION TITLE
Lora hash value of metadata different from sd-webui?

### DISCUSSION THREAD

**Discussion by 2b3d4f** [lllyasviel/Fooocus#2598] [Q&A] [answered]
When adding metadata, Lora hash values seem to be different between stable-diffusion-webui (Forge) and Fooocus. (Checkpoint hash values were the same.) Is this intended behavior?

---

### Model and Lora

- [juggernautXL_v9Rundiffusionphoto2](https://civitai.com/models/133005?modelVersionId=348913)
- [sd_xl_offset_example-lora_1.0](https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/blob/main/sd_xl_offset_example-lora_1.0.safetensors)

### Fooocus (v2.3.0)

![2024-03-22_13-00-43_9170](https://github.com/lllyasviel/Fooocus/assets/129341151/8c3751d8-2b1e-45c0-a1fb-7e0984b2c4e4)

#### hashes

```
juggernautXL_v9Rundiffusionphoto2: c9e3e68f89
sd_xl_offset_example-lora_1.0: 4852686128
```

#### metadata (a1111)

```
{
parameters:  "middle aged man, full body Steps: 30, Sampler: DPM++ 2M SDE Karras, Seed: 0, Size: 832x1216, CFG scale: 6, Sharpness: 2, ADM Guidance: "(1.5, 0.8, 0.3)", Model: juggernautXL_v9Rundiffusionphoto2, Model hash: c9e3e68f89, Performance: Speed, Scheduler: karras, Raw prompt: "middle aged man, full body", Raw negative prompt: , Lora hashes: "sd_xl_offset_example-lora_1.0: 4852686128: 0.1", Version: Fooocus v2.3.0",
metadata_scheme:  "a1111"
}
```

### stable-diffusion-webui-forge (f0.0.17v1.8.0rc-latest-276-g29be1da7)

![00008-0](https://github.com/lllyasviel/Fooocus/assets/129341151/60685750-224e-4fc9-ad4f-2997ffc59184)

#### hashes

```
juggernautXL_v9Rundiffusionphoto2: c9e3e68f89
sd_xl_offset_example-lora_1.0: 8e3e833226b3
```

#### metadata

```
{
parameters:  "middle aged man, full body, <lora:sd_xl_offset_example-lora_1.0:0.1> Steps: 30, Sampler: DPM++ 2M SDE Karras, CFG scale: 6, Seed: 0, Size: 832x1216, Model hash: c9e3e68f89, Model: juggernautXL_v9Rundiffusionphoto2, Lora hashes: "sd_xl_offset_example-lora_1.0: 8e3e833226b3", Version: f0.0.17v1.8.0rc-latest-276-g29be1da7",
metadata_scheme:  "a1111"
}
```

#### Fooocus
![meta_fooocus_civit](https://github.com/lllyasviel/Fooocus/assets/129341151/21054beb-4827-4639-a059-e2e6b9d3d698)

#### stable-diffusion-webui-forge
![meta_forge_civit](https://github.com/lllyasviel/Fooocus/assets/129341151/6b1f7bdc-06db-4747-9797-4499ccc66c03)

**Comment by mashb1t** [comment]
Thank you for reporting. Fooocus uses the same LoRA hash algorithm as A1111, which (also) wasn't working at the time. I've extracted this discussion into a new issue to check A1111/Forge soon again for potential fixes: https://github.com/lllyasviel/Fooocus/issues/2600

**Reply by 2b3d4f** [reply]
Thank you for your answer. I tried to find the cause by looking at both implementations, but it was difficult for me, so I am glad I found the cause.

**Reply by mashb1t** [reply, marked as answer]
I've double-checked, the new hashes in Fooocus are not compatible with Civitai => sticking with the old ones, but still fixed LoRA hashes in general for A1111 scheme and Civitai.
Fixed in https://github.com/lllyasviel/Fooocus/pull/2615
